### PR TITLE
Ensure Turbopack writes font optimization manifest

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Even though we don't support the legacy font optimization (what came before `next/font`) the manifest is read during builds, so we have to write an empty version of it.

Closes NEXT-2498

### CODE REVIEW

**Comment by timneutkens** [comment]
* `canary`
  * **#62079** <a href="https://app.graphite.dev/github/pr/vercel/next.js/62079?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| buildDuration | 19.9s | 20.1s | ⚠️ +112ms |
| buildDurationCached | 7.7s | 8s | ⚠️ +299ms |
| nodeModulesSize | 196 MB | 196 MB | ⚠️ +4.12 kB |
| nextStartRea..uration (ms) | 424ms | 430ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| 1068-HASH.js gzip | 30.2 kB | 30.2 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.5 kB | 53.5 kB | N/A |
| 4944-HASH.js gzip | 5.04 kB | 5.03 kB | N/A |
| 8423.HASH.js gzip | 181 B | 181 B | ✓ |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 241 B | 242 B | N/A |
| main-HASH.js gzip | 32.1 kB | 32.1 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 47.1 kB | 47.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 196 B | ✓ |
| _error-HASH.js gzip | 184 B | 183 B | N/A |
| amp-HASH.js gzip | 503 B | 504 B | N/A |
| css-HASH.js gzip | 323 B | 324 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.51 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 259 B | N/A |
| head-HASH.js gzip | 353 B | 351 B | N/A |
| hooks-HASH.js gzip | 370 B | 370 B | ✓ |
| image-HASH.js gzip | 4.21 kB | 4.2 kB | N/A |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.68 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 313 B | 314 B | N/A |
| script-HASH.js gzip | 386 B | 385 B | N/A |
| withRouter-HASH.js gzip | 309 B | 311 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 931 B | 931 B | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| _buildManifest.js gzip | 485 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| index.html gzip | 528 B | 527 B | N/A |
| link.html gzip | 541 B | 540 B | N/A |
| withRouter.html gzip | 525 B | 522 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| edge-ssr.js gzip | 94.4 kB | 94.4 kB | N/A |
| page.js gzip | 151 kB | 151 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 627 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 44.4 kB | 44.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.94 kB | 1.94 kB | ✓ |
| Overall change | 2.1 kB | 2.1 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 166 kB | 166 kB | ✓ |
| app-page-exp..prod.js gzip | 95.5 kB | 95.5 kB | ✓ |
| app-page-tur..prod.js gzip | 97.3 kB | 97.3 kB | ✓ |
| app-page-tur..prod.js gzip | 91.7 kB | 91.7 kB | ✓ |
| app-page.run...dev.js gzip | 136 kB | 136 kB | ✓ |
| app-page.run..prod.js gzip | 90.3 kB | 90.3 kB | ✓ |
| app-route-ex...dev.js gzip | 22 kB | 22 kB | ✓ |
| app-route-ex..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| app-route-tu..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| app-route-tu..prod.js gzip | 14.6 kB | 14.6 kB | ✓ |
| app-route.ru...dev.js gzip | 21.7 kB | 21.7 kB | ✓ |
| app-route.ru..prod.js gzip | 14.6 kB | 14.6 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.43 kB | 9.43 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.7 kB | 9.7 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.43 kB | 9.43 kB | ✓ |
| pages-turbo...prod.js gzip | 22 kB | 22 kB | ✓ |
| pages.runtim...dev.js gzip | 22.7 kB | 22.7 kB | ✓ |
| pages.runtim..prod.js gzip | 22 kB | 22 kB | ✓ |
| server.runti..prod.js gzip | 50 kB | 50 kB | ✓ |
| Overall change | 925 kB | 925 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 02-15-Ensure_Turbopack_writes_font_optimization_manifest | Change |
| - | - | - | - |
| 0.pack gzip | 1.55 MB | 1.55 MB | N/A |
| index.pack gzip | 103 kB | 104 kB | ⚠️ +675 B |
| Overall change | 103 kB | 104 kB | ⚠️ +675 B |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 05156c7cb8027e4dabbc5be95a10cb373d919774

**Comment by ijjk** [comment]
## Failing test suites 
Commit: d2f2dbc11202cbb2178631ab585b74e6afc4de6c

`pnpm test-dev test/e2e/app-dir/interception-route-prefetch-cache/interception-route-prefetch-cache.test.ts` (PPR)
- interception-route-prefetch-cache > runtime = edge > should render the correct interception when two distinct layouts share the same path structure

<details>
<summary>Expand output</summary>

  ● interception-route-prefetch-cache › runtime = edge › should render the correct interception when two distinct layouts share the same path structure

    thrown: "Exceeded timeout of 120000 ms for a test.
    Add a timeout value to this test to increase the timeout, if this is a long-running test. See https://jestjs.io/docs/api#testname-fn-timeout."

       6 | describe('interception-route-prefetch-cache', () => {
       7 |   function runTests({ next }: ReturnType<typeof nextTestSetup>) {
    >  8 |     it('should render the correct interception when two distinct layouts share the same path structure', async () => {
         |     ^
       9 |       const browser = await next.browser('/')
      10 |
      11 |       await browser.elementByCss('[href="/foo"]').click()

      at it (e2e/app-dir/interception-route-prefetch-cache/interception-route-prefetch-cache.test.ts:8:5)
      at runTests (e2e/app-dir/interception-route-prefetch-cache/interception-route-prefetch-cache.test.ts:91:5)
      at describe (e2e/app-dir/interception-route-prefetch-cache/interception-route-prefetch-cache.test.ts:90:3)
      at Object.describe (e2e/app-dir/interception-route-prefetch-cache/interception-route-prefetch-cache.test.ts:6:1)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/next/src/build/index.ts [hunk 1]
```diff
@@ -1465,21 +1498,21 @@ export default async function build(
...
+        //   version: 4,
+        //   routes: {},
+        //   dynamicRoutes: {},
+        //   notFoundRoutes: [],
+        //   preview: previewProps,
+        // }
+        // await writePrerenderManifest(distDir, prerenderManifest)
         // End temporary
-        throw new Error("next build doesn't support turbopack yet")
+        // throw new Error("next build doesn't support turbopack yet")
```
Comment (sokra) [inline packages/next/src/build/index.ts:1511]: If it's no longer needed, remove it

File: packages/next/src/build/index.ts [hunk 2]
```diff
@@ -1326,10 +1329,16 @@ export default async function build(
...
-      async function turbopackBuild(): Promise<never> {
+      async function turbopackBuild(): Promise<{
+        duration: number
+        buildTraceContext: undefined
+      }> {
         if (!process.env.TURBOPACK || !process.env.TURBOPACK_BUILD) {
           throw new Error("next build doesn't support turbopack yet")
         }
+        // @ts-expect-error
+        process.env.NODE_ENV = 'development'
```
Comment (sokra) [inline packages/next/src/build/index.ts:1340]: That's weird. Maybe leave a TODO why we temporarily need that

**Review by sokra — APPROVED** [review]

**Pull request merged**